sympy/geometry/line.py

```diff
@@ -2526,6 +2578,50 @@ def equation(self, x='x', y='y', z='z', k=None):
                 break
         return Tuple(*[i.subs(k, kk).as_numer_denom()[0] for i in eqs])
 
+    def bisectors(self, line):
+        """Returns the lines which are the bisections between the self line and another one.
+
+        Parameters
+        ==========
+
+        line : Line3D
+
+        Returns
+        =======
+
+        list: two Line3D instances
+
+        Examples
+        ========
+
+        >>> from sympy.geometry import Point3D, Line3D
+        >>> r1 = Line3D(Point3D(0, 0, 0), Point3D(1, 0, 0))
+        >>> r2 = Line3D(Point3D(0, 0, 0), Point3D(0, 1, 0))
+        >>> r1.bisectors(r2)
+        [Line3D(Point3D(0, 0, 0), Point3D(1, 1, 0)), Line3D(Point3D(0, 0, 0), Point3D(1, -1, 0))]
+
+        """
+
+        point = intersection(self, line)
+
+        # Three cases: Lines may intersect in a point, may be equal or may not intersect.
+        if not point:
+            return GeometryError("The lines do not intersect")
+        else:
+            if not isinstance(point[0], Point3D):
+                # Intersection is a line because both lines are coincident
+                return [self]
+
+        normalize = lambda x: Point(x).unit.args
+
+        d1 = normalize(self.direction_ratio)
+        d2 = normalize(line.direction_ratio)
+
+        bis1 = Line3D(point[0], direction_ratio=[d1[i] + d2[i] for i in range(3)])
```

Comment:
Seeing what you are doing with d1 and d2 suggest this:
```python
d1 = Point(self.direction_ratio).unit
d2 = Point(line.direction_ratio).unit
bis1 = Line3D(point[0], direction_ratio=d1 + d2)
bis2 = Line3D(point[0], direction_ratio=d1 - d2)
```

Reply:
Yes that's a clever optimization! Thanks!
If it is known that the points are 2D, should it use Point2D instead of Point? 
The same applies for the 3D case.

Reply:
SymPy does the right thing for a given number of arguments:
```python
>>> Point(2,3,4)
Point3D(2, 3, 4)
>>> Line((1,2,3),(3,2,3))
Line3D(Point3D(1, 2, 3), Point3D(3, 2, 3))
```
For this reason, too, it should only be necessary to define this function once and perhaps call it from Line2D and Line3D; in the function you could use Point and Line instead of the 3D or 2D decorated forms.

Reply:
Yes but explicit is better than implicit, right?

Reply:
See the LinearEntity class for examples of methods that are written without reference to dimension but apply to 2D and 3D. Look, too, for use of `ambient_dimension` to make sure that the case of a 2D line intersecting a 3D line is considered and handled properly.